src/Symfony/Component/ObjectMapper/Transform/MapCollection.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Transform;
+
+use Symfony\Component\ObjectMapper\Exception\MappingException;
+use Symfony\Component\ObjectMapper\ObjectMapper;
+use Symfony\Component\ObjectMapper\ObjectMapperInterface;
+use Symfony\Component\ObjectMapper\TransformCallableInterface;
+
+/**
+ * @template T of object
+ *
+ * @implements TransformCallableInterface<object, T>
+ */
+class MapCollection implements TransformCallableInterface
+{
+    public function __construct(private ObjectMapperInterface $objectMapper = new ObjectMapper())
+    {
+    }
+
+    public function __invoke(mixed $value, object $source, ?object $target): mixed
+    {
+        if (!is_iterable($value)) {
+            throw new MappingException(sprintf('The MapCollection transform expects an iterable, "%s" given.', get_debug_type($value)));
+        }
+
+        foreach ($value as &$v) {
```

Comment:
I think it'd be best to create a new array this is probably why psalm is not happy :P

Reply:
Indeed. Plus, if the iterator is not rewindable, the actual implementation can cause troubles.